deployer/develop/task/prepare_release.php

```diff
@@ -0,0 +1,201 @@
+        get('develop_release_callback');
+        add('develop_tasks', ['release_callback']);
+    }
+
+    info("Successfully prepared new release $version");
+    $rows = [];
+    foreach(get('develop_tasks') as $task) {
+        $rows[] = [
+            $task,
+            '✔️'
+        ];
+    }
+    (new Table(output()))
+        ->setHeaderTitle($version)
+        ->setHeaders(['Task', 'Status'])
+        ->setRows($rows)
+        ->render();
+
+    warning("Please check your git log and verify all automated git commits before pushing them!");
+    info("Don't forget to \"git push\"");
+})
+    ->select('alias=local')
+    ->desc('Preparing a new release');
+
+
+task('develop:release:reset', function () {
+    /* get new version */
+    if (is_null(input()->getOption('newVersion'))) {
+        throw new Exception('Missing option "version" for new release', 1711458053);
+    }
+    $version = input()->getOption('newVersion');
+
+    debug("checkout branch: " . get('develop_default_branch'));
+    tabulaRasa(true);
+    debug("remove new branch: \"release-$version\"");
+    runLocally("git branch -D \"release-$version\"");
+
+    info("Reset back to " . get('develop_default_branch'));
+})
+    ->select('alias=local')
+    ->desc('Delete a new release');
+
+function checkPreconditions() {
+    $version = get('newVersion');
+    if (!preg_match(get('develop_semver_regex'), $version, $matches)) {
+        throw new Exception("Given version '$version' not matching semantic version constraints", 1711458465);
+    }
+
+    runLocally("composer validate --working-dir " . get('composer_path_app'));
+}
+
+
+function tabulaRasa(bool $force = false): void {
+    info("[step] tabula rasa");
+    if (!$force) {
+        $modifiedFiles = runLocally("git status -uno -s");
+        if ($modifiedFiles) {
+            throw new Exception("Please commit modified files before starting a new release", 1711460221);
+        }
+
+        debug("checkout branch: " . get('develop_default_branch'));
+        runLocally('git pull');
+        runLocally('git remote prune origin');
+    }
+
+    $additional = $force ? ' --force' : '';
+    runLocally('git checkout ' . get('develop_default_branch') . $additional);
+
+    /*  */
+    debug("install dependencies");
+    runLocally("composer install --working-dir " . get('composer_path_app'));
+
+    /* additional tabula rasa tasks */
+    get('develop_tabula_rasa_callback');
+}
+
+function startNewRelease(): void {
+    info("[step] start new release");
+    $version = get('newVersion');
+    /* new git branch */
+    debug("create new branch: \"release-$version\"");
+    runLocally("git checkout -b \"release-$version\"");
+
+    /* new release version */
+    debug("set new version: \"{$version}-RC\"");
+    runLocally("composer config version {$version}-RC --working-dir " . get('composer_path_app'));
+    runLocally("composer update nothing -q --no-scripts --no-plugins --working-dir " . get('composer_path_app'));
```

Comment:
This throws errors in drupal context:

```
vendor/bin/dep develop:release:prepare local --newVersion=3.7.0
task develop:release:prepare
[local] info The following steps will be executed:
 - tabula_rasa 
 - start_new_release 
 - composer_update_ci
 - composer_update_app
[local] info Which steps shall be executed?
[local] composer_update_ci (default: yes) 
[local] composer_update_app (default: yes)   
[local] info Start preparation ...
[local] info [step] tabula rasa
[local] info [step] start new release
[local]  error  in prepare_release.php on line 130:
[local] run composer update nothing -q --no-scripts --no-plugins --working-dir ./app
[local] err Your requirements could not be resolved to an installable set of packages.
[local] exit code 2 (Misuse of shell builtins)
```

When removing the `--no-plugins` it works.

Also adding `--no-scripts` causes the robots.txt not to be adjusted resulting in a commit with a plain robots.txt without our adjustments in the commit. Later on this gets fixed when updating the app "normally" (without `--no-scripts`), but it's a bit bad nevertheless.

Maybe we should parameterize this like so:

```
runLocally("composer update nothing -q " . get('composer_app_release_options') . " --working-dir " . get('composer_path_app'));
```

with the default value `--no-scripts --no-plugins` in the `set.php`?

Reply:
I added this configuration for now, but I'm not so happy with this, because this is just the composer function to update the release number and not all kind of composer release options. Maybe need some better naming or so.

Reply:
Yep, the naming was kind of a shot in the dark 😅️ Thanks for adding.